java/ql/src/Likely Bugs/Resource Leaks/CloseReader.ql

```diff
@@ -17,17 +17,14 @@ import CloseType
 
 predicate readerType(RefType t) {
   exists(RefType sup | sup = t.getASupertype*() |
-    sup.hasName("Reader") or
-    sup.hasName("InputStream") or
+    sup.hasName(["Reader", "InputStream"]) or
     sup.hasQualifiedName("java.util.zip", "ZipFile")
   )
 }
 
 predicate safeReaderType(RefType t) {
   exists(RefType sup | sup = t.getASupertype*() |
-    sup.hasName("StringReader") or
-    sup.hasName("ByteArrayInputStream") or
-    sup.hasName("StringInputStream")
```

Comment:
I have removed `StringInputStream` because it is not clear which class this is supposed to match. The JDK had a class with this name , but it looks like it was removed before the JDK 1.0 release, see [JDK-1210406](https://bugs.openjdk.java.net/browse/JDK-1210406).
There is still [`StringBufferInputStream`](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/io/StringBufferInputStream.html), but that class is deprecated so it might not be worth including that here.

(In general it might be good to check for qualified class names here, let me know if I should change that as well.)

Reply:
Done in commit 3d1969f; I have also added tests for `CloseWriter.ql` based on on the ones for `CloseReader.ql`.